packages/ses/index.d.ts

```diff
@@ -0,0 +1,112 @@
+type safetyPreference = 'safe' | 'unsafe';
+
+type LockdownOpts = {
+  fringeSet?: {
+    add?: Function,
+    has?: Function,
+  },
+  dateTaming?: safetyPreference,
+  errorTaming?: safetyPreference,
+  mathTaming?: safetyPreference,
+  regExpTaming?: safetyPreference,
+
+  [extraOption:string]?: any,
+};
+
+declare var lockdown = (LockdownOpts) => void;
+
+declare var harden = (any) => any;
+
+// Shared space by Realm
+type FullSpecifier = string;
+type ModuleNamespace = object;
+
+declare class StaticModuleRecord {
+  constructor (moduleSource: string, moduleLocation:string);
+  // intend to add reflection of import/export bindings
+
+  // needs to allow duplicates
+  staticImports(): {specifier: string, exportNames}[];
+}
+
+type CompartmentConstructorOptions = {
+  // JF has a better way for:
+  //   randomHook(): number; // Use for Math.random()
+  //   nowHook(): number; // Use for both Date.now() and new Date(),
+  // Supplied during Module instance creation instead of option
+  //   hasSourceTextAvailableHook(scriptOrModule): bool; // Used for censorship
+  resolveHook(name: string, referrer: FullSpecifier): FullSpecifier
+
+  // timing
+  // importHook delegates to importNowHook FIRST,
+  // they share a memo cache
+  // order to ensures importNow never allows async value of import
+  // to be accessed prior to any attached promise
+  importHook(fullSpec: FullSpecifier): Promise<StaticModuleRecord>;
+  importNowHook(fullSpec: FullSpecifier): StaticModuleRecord?;
+
+  // copy own props after return
+  importMetaHook(fullSpec: FullSpecifier): object
+
+  // e.g.: 'fr-FR' - Affects appropriate ECMA-402 APIs within Compartment
+  localeHook(): string;
+  // This is important to be able to override for deterministic testing and such
+  localTZAHook(): string;
+
+  // determines if the fn is acting as an "eval" function
+  isDirectEvalHook(evalFunctionRef: any): boolean;
+
+  // prep for trusted types non-string
+  canCompileHook(source: any, {
+    evaluator: functionRef, // can be a value from isDirectEvalHook
+    isDirect?: boolean
+  }): boolean; // need to allow mimicing CSP including nonces
```

Comment:
We have not implemented any of these, and might not. I’m in favor of removing these for now and add them back when the spec is more firm. We’re not going to pursue these in the initial Compartments proposal since they are not necessary for containment, but would be good to add when it’s a better time to look into full host virtualization.